packages/monitor-v2/src/monitor-og/oSnapAutomation.ts

```diff
@@ -374,6 +452,60 @@ const submitProposals = async (
   }
 };
 
+const submitDisputes = async (logger: typeof Logger, proposals: DisputableProposal[], params: MonitoringParams) => {
+  assert(params.signer !== undefined, "Signer must be set to dispute proposals.");
+  const disputerAddress = await params.signer.getAddress();
+
+  for (const proposal of proposals) {
+    const oo = await getOo(params);
+
+    // Approve bond based on passed proposal parameters.
+    await approveBond(
+      params.provider,
+      params.signer,
+      proposal.parameters.currency,
+      proposal.parameters.bond,
+      oo.address
+    );
+
+    // Prepare potential error message for simulating/disputing.
+    const disputeError =
+      "Dispute submission on assertionId " +
+      proposal.event.args.assertionId +
+      " related to proposalHash " +
+      proposal.event.args.proposalHash +
+      " posted on oSnap module " +
+      proposal.event.address +
+      " for Snapshot space " +
+      proposal.parameters.parsedRules.space;
+
+    // Check that dispute submission would succeed.
+    try {
+      await oo.callStatic.disputeAssertion(proposal.event.args.assertionId, disputerAddress, {
+        from: disputerAddress,
+      });
+    } catch (error) {
+      assert(error instanceof Error, "Unexpected Error type!");
+      throw new Error(`${disputeError} would fail: ${error.message}`);
```

Comment:
Instead of throwing an error, it might make sense to just logger.error (to emit an error) and continue to the next proposal. That way, one dispute breaking won't stop the bot from processing the other disputes.

Reply:
I'll do this in a separate PR as this approach an also be used for proposer